decidim/decidim
# Allow custom session store and remove samesite protection config

#### :tophat: What? Why?
If the environment config has already configured the session store, do not override this configuration.

This also renames the initializer for better alignment with the naming conventions and also removes the unnecessary same site protection config since Rails 6.1:

![Same site configuration defaults for Rails 6.1](https://github.com/decidim/decidim/assets/864340/f39e91a6-5351-4f5d-9c1e-f6ed5118e03d)

https://edgeguides.rubyonrails.org/configuring.html#default-values-for-target-version-6-1

#### Testing
- Configure the session store in the Rails environment configurations, e.g. change to cache store or change the session storage options
- Restart the server and see that Decidim is no longer overriding these configurations

As an example use case, I needed to change the `SameSite` protection for the session cookie due to it breaking the user redirection back to the correct page after returning from the authentication service. Some implementors could also want to use a different session store than the cookie store.
